### PR TITLE
Move Data internal methods to internal headers

### PR DESCRIPTION
**Pull request by jinquan2019**
Things to consider before you submit the PR:

* [ ] Has `CHANGELOG.md` been updated?
* [ ] Are tests passing locally?
* [X] Are the files formatted correctly?
* [ ] Did you add unit tests?
* [ ] Did you check UI tests on the sample app? They are not executed on CI.
* [ ] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

Move Data module internal methods marked as `// TODO move to internal` from public header to internal header.

### CODE REVIEW

File: AppCenterData/AppCenterData/Internal/Model/MSDataErrorInternal.h [hunk 1]
```diff
@@ -0,0 +1,16 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#import "MSDataError.h"
+
+@interface MSDataError ()
```
Comment (clpolet) [inline AppCenterData/AppCenterData/Internal/Model/MSDataErrorInternal.h]: You need the nullability declaration in this file. Please use `NS_ASSUME_NONNULL_BEGIN` and `NS_ASSUME_NONNULL_END`.

File: AppCenterData/AppCenterData/Internal/Model/MSPageInternal.h [hunk 2]
```diff
@@ -0,0 +1,26 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#import "MSPage.h"
+
```
Comment (clpolet) [inline AppCenterData/AppCenterData/Internal/Model/MSPageInternal.h]: You need the nullability declaration in this file too. Please use `NS_ASSUME_NONNULL_BEGIN` and `NS_ASSUME_NONNULL_END`.

File: AppCenterData/AppCenterData/Internal/Model/MSPaginatedDocumentsInternal.h [hunk 3]
```diff
@@ -1,8 +1,8 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
-#import "MSPage.h"
-#import "MSReadOptions.h"
+#import "MSPageInternal.h"
```
Comment (clpolet) [inline AppCenterData/AppCenterData/Internal/Model/MSPaginatedDocumentsInternal.h]: You can make the MSPage import a class forwarding declaration instead.

File: AppCenterData/AppCenterData.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -856,6 +860,7 @@
 			developmentRegion = English;
 			hasScannedForEncodings = 0;
 			knownRegions = (
+				English,
```
Comment (jaelim-ms) [inline AppCenterData/AppCenterData.xcodeproj/project.pbxproj]: What's this change?

File: AppCenterData/AppCenterData/Internal/Model/MSDataErrorInternal.h [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+/**
+ * Create an instance with error object.
+ *
+ * @param error An error object.
+ *
+ * @return A new `MSDataError` instance.
+ */
+- (instancetype)initWithError:(NSError *)error;
+@end
```
Comment (jwallra) [inline AppCenterData/AppCenterData/Internal/Model/MSDataErrorInternal.h]:
```suggestion

@end
```

File: AppCenterData/AppCenterData.xcodeproj/project.pbxproj (hunk 4 above)
Comment (clpolet) [reply AppCenterData/AppCenterData.xcodeproj/project.pbxproj]: Probably introduced by Xcode itself.

Comment (jaelim-ms) [reply AppCenterData/AppCenterData.xcodeproj/project.pbxproj]: I removed that. I wonder which version of Xcode was used for this PR.

**Pull request merged**